src/tir/transforms/narrow_datatype.cc

```diff
@@ -253,6 +253,23 @@ class DataTypeRewriter : public StmtExprMutator {
     return StmtExprMutator::VisitExpr_(op);
   }
 
+  PrimExpr VisitExpr_(const RampNode* op) final {
+    PrimExpr base = VisitExpr(op->base);
+    PrimExpr stride = VisitExpr(op->stride);
+    if (base.same_as(op->base) && stride.same_as(op->stride)) {
+      return GetRef<PrimExpr>(op);
+    } else {
+      if (base.dtype().is_int()) {
+        ICHECK(stride.dtype().is_int()) << "Ramp base is int but stride is " << stride.dtype();
```

Comment:
I think we can simply assume that base and stride should be of integer types. However, I also noticed that in 

https://github.com/apache/tvm/blob/22c488e3a829ad700de6547be6096fb2d1f02e81/src/tir/ir/expr.cc#L705

Such assumptions are not checked. I am a bit curious if there will be, say `base/stride` in float types?